src/EditorFeatures/Core/Implementation/IntelliSense/SignatureHelp/Controller.cs

```diff
@@ -98,37 +76,41 @@ internal override void OnModelUpdated(Model modelOpt)
             }
         }
 
-        private void StartSession(IList<ISignatureHelpProvider> providers, SignatureHelpTriggerInfo triggerInfo)
+        private void StartSession(SignatureHelpTrigger trigger)
         {
             AssertIsForeground();
             VerifySessionIsInactive();
 
             this.sessionOpt = new Session(this, Presenter.CreateSession(TextView, SubjectBuffer, null));
-            this.sessionOpt.ComputeModel(providers, triggerInfo);
+            this.sessionOpt.ComputeModel(trigger);
         }
 
-        private IList<ISignatureHelpProvider> GetProviders()
+        private SignatureHelpService GetSignatureHelpService()
         {
-            this.AssertIsForeground();
+            Workspace workspace;
+            if (!Workspace.TryGetWorkspace(this.SubjectBuffer.AsTextContainer(), out workspace))
+            {
+                Trace.WriteLine("Failed to get a workspace, cannot have a signature help service.");
+                return null;
+            }
 
-            var snapshot = this.SubjectBuffer.CurrentSnapshot;
-            var currentContentType = snapshot.ContentType;
+            return workspace.Services.GetLanguageServices(this.SubjectBuffer).GetService<SignatureHelpService>();
```

Comment:
This seems weird.  Why are we usign the buffer instead of just getting the document and working off of that?  This seems like a strange pattern that i havn'et seen elsewhere.